dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/DatasetDao.java

```diff
@@ -89,6 +91,11 @@ public List<Dataset> findAll() {
         return em.createQuery("select object(o) from Dataset as o order by o.id", Dataset.class).getResultList();
     }
 
+    public List<Dataset> findNotIndexedAfterEmbargo() {
+        TypedQuery<Dataset> typedQuery = em.createQuery("select d from Dataset d, DvObject o where d.id = o.id and d.embargoDate < :actualTimestamp and d.embargoDate > o.indexTime", Dataset.class);
```

Comment:
Probably this query could be written in more concise way, as Dataset extends DvObject, but obviously it's ok as it is.